releasenotes/notes/vf2-post-layout-max-trials-98b1c736e2e33861.yaml

```diff
@@ -0,0 +1,8 @@
+---
+features:
+  - |
+    Added a new parameter ``max_trials`` to pass :class:`~.VF2PostLayout`
+    which, when specified,limits the number of trials performed when
+    searching for the best layout. Previously, all possible layouts were
+    always considered, which is not suitable when performing layout of
+    multiple connected components on large devices.
```

Comment:
Ah, yeah it's probably worth noting that call limit could cause a specific VF2 trial to bail. This would just result in a `None` value for the particular result of `next(<iter>)` though, correct? Wouldn't we keep exploring additional VF2 mappings after this?

Reply:
If the `vf2_mapping()` reaches a set reaches the call limit value it exits the iterator instead of returning another layout. It effectively ends the search when we reach the call limit

Reply:
Never mind, I see that we store the number of calls in the iterator itself in Rustworkx, so once we hit the call limit, the iterator won't give us anything new back.

Reply:
Updated in 1a9639c824ff2161e1291ecfe49dbdabb462e272. Thoughts?